### PR TITLE
Can a contract send data on a different's contract IBC

### PR DESCRIPTION
**Pull request by GNaD13**
#660
Add more test IBC edge cases

### CODE REVIEW

File: x/wasm/relay_test.go [hunk 1]
```diff
@@ -356,6 +356,134 @@ func TestContractCanEmulateIBCTransferMessageWithTimeout(t *testing.T) {
...
+	myContractA2.contractAddr = myContractAddr2.String()
+
+	path := wasmibctesting.NewPath(chainA, chainB)
+	path.EndpointA.ChannelConfig = &ibctesting.ChannelConfig{
+		PortID:  chainA.ContractInfo(myContractAddr1).IBCPortID,
+		Version: ibctransfertypes.Version,
+		Order:   channeltypes.UNORDERED,
+	}
+	path.EndpointB.ChannelConfig = &ibctesting.ChannelConfig{
+		PortID:  ibctransfertypes.PortID,
```
Comment (alpe) [inline x/wasm/relay_test.go:395]: 👍 no contract needed on this side

**Review by alpe — APPROVED** [review]
Another scenaroi covered by tests. Thanks a lot for this! 💯 
Not sure if the second test adds value. Feel free to keep or remove it.

**Comment by GNaD13** [comment]
> Another scenaroi covered by tests. Thanks a lot for this! 💯 Not sure if the second test adds value. Feel free to keep or remove it.

Thank you so much for your reviews.
And moreover, as you said in #660 , I think that we have 3 more case which is
- Can a contract send data to a ICS20 channel (local loop)?
- Can a contract send data on a different's contract IBC counterparty channel ?
- Can a contract send data to a different's contract IBC channel (local loop)?
About these 3 case, I can't imagine how they working. Can you give me more detail about these scenarios please?

File: x/wasm/relay_test.go [hunk 2]
```diff
@@ -356,6 +356,134 @@ func TestContractCanEmulateIBCTransferMessageWithTimeout(t *testing.T) {
...
+	timeout := uint64(chainB.LastHeader.Header.Time.Add(time.Hour).UnixNano()) // enough time to not timeout
+	receiverAddress := chainB.SenderAccount.GetAddress()
+	coinToSendToB := sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100))
+
+	// start transfer from chainA - A2 to chainB via IBC channel
+	startMsg := &types.MsgExecuteContract{
+		Sender:   chainA.SenderAccount.GetAddress().String(),
+		Contract: myContractAddr2.String(),
+		Msg: startTransfer{
+			ChannelID:    path.EndpointB.ChannelID,
```
[the remark this answers is not in the document]
Comment (GNaD13) [reply x/wasm/relay_test.go:476]: Got it!!

[the remark this answers is not in the document]
Comment (GNaD13) [reply x/wasm/relay_test.go]: I've removed this case, tysm :3
